nfs/nfs.go

```diff
@@ -0,0 +1,90 @@
+package nfs
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/longhorn/go-common-libs/types"
+	"github.com/pkg/errors"
+)
+
+const (
+	nfsmountGlobalSection  = "NFSMount_Global_Options"
+	nfsmountDefaultVersion = "Defaultvers"
+)
+
+// GetSystemDefaultNFSVersion reads the system default NFS version. This config can be overridden by nfsmount.conf under
+// configPrefix. If configDir is empty, it will be /etc by default. If the nfsmount.conf is absent, or the default
+// version is not overridden in the configuration file, the returned version will fallback to defaultMajor and defaultMinor.
+func GetSystemDefaultNFSVersion(configDir string, defaultMajor, defaultMinor int) (major, minor int, err error) {
+	if configDir == "" {
+		configDir = types.SysConfigDirectory
+	}
+
+	configMap, err := getSystemNFSMountConfigMap(configDir)
+	if os.IsNotExist(err) {
+		return defaultMajor, defaultMinor, nil
```

Comment:
If nfsmount.conf is absent, the NFS client actually negotiates with the server from the latest supported version, and it would be 4 currently. Caller should be aware of default protocol version if we return an error here.

Reply:
To make it simpler, will remove the default versions from parameter. Caller is responsible to handle the configuration absence.